account_tax_analysis/models/account_move_line.py

```diff
@@ -20,9 +20,15 @@ def _compute_analysis_tax(self):
         for company in companies:
             lines = self.filtered(lambda s: s.company_id == company)
             lang_lines = lines
-            if company.partner_id.lang:
+            if (company.partner_id.lang and
+                    company.partner_id.lang != self.env.user.partner_id.lang):
                 lang_lines = lines.with_context(lang=company.partner_id.lang)
             for line, lang_line in zip(lines, lang_lines):
-                line.analysis_tax = (
-                    lang_line.tax_line_id.description or
-                    ', '.join(sorted(lang_line.tax_ids.mapped('description'))))
+                if lang_line.tax_line_id.description:
+                    line.analysis_tax = lang_line.tax_line_id.description
+                else:
+                    descriptions = lang_line.tax_ids.mapped('description')
+                    if False in descriptions:
+                        # remove all False from the descriptions list
+                        descriptions = filter(lambda c: c, descriptions)
```

Comment:
no no! something must be displayed tax name or whatever. Removing it from the list is dangerous.

Reply:
`description or name`

Reply:
hm, no, please use `name` always, it's unique and required.

Reply:
I will change it to be `tax.description or tax.name`, since name is required

Reply:
no, use `name` always please, not description.

Since `description` is not guaranteed to be unique it could create confusion.